ui2/src/pages/custody/Account.tsx

```diff
@@ -28,9 +29,19 @@ const AccountComponent: React.FC<RouteComponentProps & ServicePageProps<Service>
   const cid = contractId.replace('_', '#');
 
   const { contracts: accounts, loading: accountsLoading } = useStreamQueries(AssetSettlementRule);
+  const { contracts: allocatedAccounts, loading: allocatedAccountsLoading } = useStreamQueries(AllocationAccountRule);
   const { contracts: deposits, loading: depositsLoading } = useStreamQueries(AssetDeposit);
   const { contracts: assets, loading: assetsLoading } = useStreamQueries(AssetDescription);
 
+  const allAccounts = useMemo(
+    () => accounts.map(a => {
+      return { account: a.payload.account, contractId: a.contractId.replace('#', '_') }
```

Comment:
I've noticed this before as well -- why replace `#` with `_`?

Reply:
That's a good question - I presume its because we're using a `<HashRouter>` in our code base that if the `Contract Id` contains a `#` then it will break the router as there'll be two hashes in the URL. To be transparent, I haven't tested this theory. This logic was originally implemented by @GeorgSchneider and I've simply ensured its consistent when generating URLs with `Contract Id`s.

Reply:
Indeed, there's a problem with the router when there's a `#` in the URL. Hence, the replacement here and back again in the page being navigated to.